### PR TITLE
[layer/model] Reduce dependence on LayerV1

### PR DESCRIPTION
**Pull request by kparichay**
Reduce dependence on LayerNode()->getObject() which returns LayerV1
object.

Reduce the usage of getObject() for optimizer.
This patch changes optimizer interface to work on each weight
individually and adds a getWeightObject() for the LayerContext.
This allows removal of getObject() from the neural network class for
backwarding.

This patch also disable dynamic_training_optimization.

Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>

### CODE REVIEW

**Comment by zhoonit** [comment]
Could you clarify how many commit should be reviewed? If it's directly depending on #1332, it's looks like there is three commits.

**Comment by kparichay** [comment]
> Could you clarify how many commit should be reviewed? If it's directly depending on #1332, it's looks like there is three commits.

Yes, there are 3 commits to be reviewed.

File: nntrainer/layers/layer_node.cpp [hunk 1]
```diff
@@ -335,29 +332,46 @@ void LayerNode::finalize() {
  * @brief     Forward Propagation of a layer
  */
 void LayerNode::forwarding(bool training) {
+  if (layerv1)
+    layerv1->forwarding(training);
+      else
```
Comment (zhoonit) [inline nntrainer/layers/layer_node.cpp]: How clang-format formats is quite weird 🤣

**Review by zhoonit — APPROVED** [review]
LGTM

File: nntrainer/layers/layer_node.cpp (hunk 1 above)
Comment (kparichay) [reply nntrainer/layers/layer_node.cpp]: yeah, i dont know why.

**Comment by zhoonit** [comment]
Android build failing. I wonder this is intended?

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Pull request merged**